pkg/awsclient/elb_helper.go

```diff
@@ -0,0 +1,170 @@
+package awsclient
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elb"
+)
+
+// AWSLoadBalancer a handy way to return information about an ELB
+type AWSLoadBalancer struct {
+	ELBName   string // Name of the ELB
+	DNSName   string // DNS Name
+	DNSZoneId string // Zone ID
+}
+
+// CreateClassicELB creates a classic ELB in Amazon, as in for management API endpoint.
+// inputs are the name of the ELB, the availability zone(s) and subnet(s) the
+// ELB should attend, as well as the listener port.
+// The port is used for the instance port and load balancer port
+// Return is the (FQDN) DNS name from Amazon, and error, if any.
+func (c *awsClient) CreateClassicELB(elbName string, subnets []string, listenerPort int64) (*AWSLoadBalancer, error) {
+	fmt.Printf("  * CreateClassicELB(%s,%s,%d)\n", elbName, subnets, listenerPort)
+	i := &elb.CreateLoadBalancerInput{
+		LoadBalancerName: aws.String(elbName),
+		Subnets:          aws.StringSlice(subnets),
+		//AvailabilityZones: aws.StringSlice(availabilityZones),
+		Listeners: []*elb.Listener{
+			{
+				InstancePort:     aws.Int64(listenerPort),
+				InstanceProtocol: aws.String("tcp"),
+				Protocol:         aws.String("tcp"),
+				LoadBalancerPort: aws.Int64(listenerPort),
+			},
+		},
+	}
+	_, err := c.CreateLoadBalancer(i)
+	if err != nil {
+		return &AWSLoadBalancer{}, err
+	}
+	fmt.Printf("    * Adding health check (HTTP:6443/)\n")
+	err = c.addHealthCheck(elbName, "HTTP", "/", 6443)
+	if err != nil {
+		return &AWSLoadBalancer{}, err
+	}
+	// Caller will need the DNS name and Zone ID for the ELB (for route53) so let's make a handy object to return, using the
+	_, awsELBObj, err := c.DoesELBExist(elbName)
```

Comment:
Recommendation: Occasionally objects take a while to become available via the API. So if this describe is called immediately after the create, it may not come back as available. In the account operator, we've implemented a retry so that the API inconsistenciesfailui don't lead to accidental failure prematurely.

Reply:
The retry happens in the apischeme controller, and not here in the library. That code is to be committed.

Reply:
There's another one on line 24! Both have been removed in my working copy that will be merged once Sam's code is merged.